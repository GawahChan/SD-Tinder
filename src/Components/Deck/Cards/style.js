import styled from 'styled-components';
import { motion } from 'framer-motion';

export const CardsContainer = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    min-height: 100vh;
`
// eslint-disable-next-line 
export const CardContainer = styled(motion.div).attrs(props => ({
    style: {
        boxShadow: props.actionColour === 'liked' ? '0px 0px 20px #15ff00' :
            props.actionColour === 'disliked' ? '0px 0px 20px #ff0000' :
                props.actionColour === 'superliked' ? '0 0 20px #02fdfd' : '0px 0px 10px #080808',

        border: props.actionColour === 'liked' ? '1px solid #15ff00' :
            props.actionColour === 'disliked' ? '1px solid #ff0000' :
                props.actionColour === 'superliked' ? '1px solid #02fdfd' : 'none'
    }
}))
    `
    position: absolute;
    border-radius: 10px;
    overflow: hidden;

    width: 320px;
    height: 500px;
        
    transition: box-shadow 0.5s, border 0.5s;
    will-change: transform;
`

export const CardContents = styled(motion.div)`
    width: 100%;
    height: 100%;

    display: flex;
    flex-direction: column;
    justify-content: flex-end; 

    background-image: ${props => {
        return (
            `linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0), rgba(0,0,0,0.5), rgba(0,0,0,1)),
            url(${require(`../../../Common/Profile/${props.img}.jpg`)})`
        )

    }};
    background-position: top;
    background-size: cover;

    transition: box-shadow 0.5s;
    will-change: transform;
`

export const CardContent = styled.div`
    padding: 1rem;
    color: #DDDDDD;
`
// eslint-disable-next-line 
export const CardOverlay = styled.h1.attrs(props => ({
    style: {
        color: props.actionColour === 'LIKED' ? '#15ff00' :
            props.actionColour === 'DISLIKED' ? '#ff0000' :
                props.actionColour === 'SUPERLIKED' ? '#02fdfd' : 'black'
    }
}))
    `
    height: 30%;
    width: 100%;

    display: flex;
    justify-content: center;
    align-items: center;

    font-size: 3rem;
    font-weight: 500;

`
export const CardTitle = styled.h1`
    margin: 0;
    font-weight: 400;
    font-size: 2rem;
`

export const CardSubtitle = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
`

export const CardInfoButton = styled(motion.div)`
    background-image: url(${require(`../../../Common/Icons/info.svg`)});
    background-size: cover;

    height: 25px;
    width: 25px;
    cursor: pointer;
    outline: 0;
`

export const CardButtonContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 5px; 
`

export const CardLikedButton = styled(motion.div)`
    background-image: url(${require(`../../../Common/Icons/liked.svg`)});
    background-size: cover;
    height: 50px;
    width: 50px;
    margin: 10px;
    cursor: pointer;
    outline: 0;
`

export const CardDislikedButton = styled(CardLikedButton)`
    background-image: url(${require(`../../../Common/Icons/disliked.svg`)});

`

export const CardSuperlikedButton = styled(CardLikedButton)`
    background-image: url(${require(`../../../Common/Icons/superliked.svg`)});

`