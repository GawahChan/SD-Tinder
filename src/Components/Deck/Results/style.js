import styled from 'styled-components';

export const ResultsContainer = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    min-height: 100vh;

    text-align: center;
    margin: 30px;
`

export const ResultDisplay = styled.div`
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
`

export const ResultHeader = styled.h1`
    color: white;
    font-weight: 400;
    font-size: 2.5rem;
`

export const ResultTitle = styled(ResultHeader)`
    font-size: 1.5rem;
`

export const ResultSubTitle = styled(ResultHeader)`
    font-size: 1rem;
    margin: 0;
`

export const ResultProfile = styled.div`
    background-image: ${props => {
        return (
            `linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0), rgba(0,0,0,0.6), rgba(0,0,0,1)),
            url(${require(`../../../Common/Profile/${props.img}.jpg`)})     `
        )
    }};
    
    border-radius: 10px;
    box-shadow: 1px 1px 8px #333;
    margin: 10px;

    background-size: cover;
    background-position: top;
    overflow: hidden;

    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    
    min-width: 250px;
    min-height: 250px;
`