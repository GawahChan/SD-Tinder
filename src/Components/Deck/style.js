import styled from "styled-components";
import { motion } from "framer-motion";

export const CardsContainer = styled(motion.div)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  flex: 1;
`;
// eslint-disable-next-line
export const CardContainer = styled(motion.div).attrs(props => ({
  style: {
    boxShadow:
      props.actionColour === "liked"
        ? "0px 0px 20px #15ff00"
        : props.actionColour === "disliked"
        ? "0px 0px 20px #ff0000"
        : props.actionColour === "superliked"
        ? "0 0 20px #02fdfd"
        : "0px 0px 10px #080808",

    border:
      props.actionColour === "liked"
        ? "1px solid #15ff00"
        : props.actionColour === "disliked"
        ? "1px solid #ff0000"
        : props.actionColour === "superliked"
        ? "1px solid #02fdfd"
        : "none"
  }
}))`
  position: absolute;
  border-radius: 10px;
  overflow: hidden;
  cursor: pointer;

  width: 320px;
  height: 480px;

  transition: box-shadow 0.5s, border 0.5s;
  will-change: transform;
`;

export const CardContents = styled(motion.div)`
  width: 100%;
  height: 100%;

  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  background-image: ${props =>
    `url(${require(`../../Common/Profile/${props.img}.jpg`)})`};
  background-position: top;
  background-size: cover;

  transition: box-shadow 0.5s;
  will-change: transform;
`;

export const CardContent = styled.div`
  padding: 10px;
  box-sizing: border-box;
  color: #dddddd;
  background: linear-gradient(
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0.7),
    rgba(0, 0, 0, 0.8),
    rgba(0, 0, 0, 1)
  );
`;

// eslint-disable-next-line
export const CardOverlay = styled.h1.attrs(props => ({
  style: {
    color:
      props.actionColour === "LIKED"
        ? "#15ff00"
        : props.actionColour === "DISLIKED"
        ? "#ff0000"
        : props.actionColour === "SUPERLIKED"
        ? "#02fdfd"
        : "black"
  }
}))`
  height: 30%;
  width: 100%;

  display: flex;
  justify-content: center;
  align-items: center;

  font-size: 3rem;
  font-weight: 500;
`;

export const CardSubtitleContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  box-sizing: border-box;
  padding-top: 5px;
  padding-bottom: 5px;
`;

export const CardButtonContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Container = styled.div`
  box-sizing: border-box;
  padding: 5px;
`;
