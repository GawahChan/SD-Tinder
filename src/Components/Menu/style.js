import styled, { css } from "styled-components";
import { Link } from "react-router-dom";
import { motion } from "framer-motion";

export const MenuContainer = styled.div`
  position: fixed;
  z-index: 1;
  background-color: none;

  ${props =>
    props.displayMenu &&
    css`
      background-color: rgba(250, 250, 250, 0.5);
      height: 100%;
      width: 100%;
    `}
`;

export const MenuPanelContainer = styled(motion.div)`
  position: absolute;
  height: 100vh;
  width: 320px;

  background-color: rgba(250, 250, 250, 0.9);
  left: -350px;

  ${props =>
    props.displayMenu &&
    css`
      left: 0px;
    `}

  transition: left 1s;
`;

export const MenuPanel = styled.div`
  box-sizing: border-box;

  padding: 75px 5px 10px 5px;
`;

export const MenuLink = styled(Link)`
  text-decoration: none;
`;

export const Container = styled.div`
  box-sizing: border-box;
  padding: 5px;
`;
