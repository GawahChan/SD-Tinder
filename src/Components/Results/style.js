import styled from "styled-components";

export const ResultsPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;

  min-height: 100vh;

  text-align: center;
  box-sizing: border-box;
  padding: 30px;
`;

export const ResultsContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  box-sizing: border-box;
  padding: 20px;
  flex: 1;
`;

export const ResultContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  box-sizing: border-box;
  padding: 10px;
  flex: 1;
`;

export const ResultDisplay = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-items: center;
  flex: 1;
`;

export const ResultProfile = styled.div`
  background-image: ${props => {
    return `linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0), rgba(0,0,0,0.2), rgba(0,0,0,1)),
            url(${require(`../../Common/Profile/${props.img}.jpg`)})     `;
  }};

  border-radius: 10px;
  border: ${props => props.theme.border};
  box-shadow: ${props => props.theme.shadow};
  margin: 10px;

  background-size: cover;
  background-position: top;
  overflow: hidden;

  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  min-width: 250px;
  min-height: 250px;
`;

export const Container = styled.div`
  box-sizing: border-box;
  padding: 10px;
`;
