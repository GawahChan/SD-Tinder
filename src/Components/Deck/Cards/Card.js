import React, { Component } from 'react';
import { CardContainer } from './style';
import CardDisplay from './CardDisplay';

class Card extends Component {
    constructor(props) {
        super(props);
        this.state = {
            action: '',
            removeCard: false
        }
    }

    animateAction = (id, action) => {
        let handleCardAnimation =
            action === 'liked' ? { x: 400, opacity: 0 } :
                action === 'disliked' ? { x: -400, opacity: 0 } :
                    action === 'superliked' ? { y: -400, opacity: 0 } :
                        { x: 0, y: 0 }

        this.setState({ action, removeCard: handleCardAnimation })

        this.props.handleAction(id, action);
    }

    handleDrag = (info) => {
        let xPosition = info.point.x;
        let yPosition = info.point.y;

        let action =
            xPosition > 30 ? 'liked' :
                xPosition < -30 ? 'disliked' :
                    yPosition < -30 ? 'superliked' : '';

        this.setState({ action });
    }

    handleDragEnd = (id, info) => {
        let xPosition = info.point.x;
        let yPosition = info.point.y;

        return (
            xPosition > 80 ? this.animateAction(id, 'liked') :
                xPosition < -80 ? this.animateAction(id, 'disliked') :
                    yPosition < -80 ? this.animateAction(id, 'superliked') :
                        this.setState({ action: '' })
        );
    }
    render() {
        return (
            <CardContainer
                drag
                dragElastic={0.8}
                dragConstraints={{ top: 0, right: 0, bottom: 0, left: 0 }}
                onDrag={(e, info) => this.handleDrag(info)}
                onDragEnd={(e, info) => this.handleDragEnd(this.props.stuntDouble.id, info)}
                animate={this.state.removeCard}
                actionColour={this.state.action}
            >
                <CardDisplay
                    stuntDouble={this.props.stuntDouble}
                    animateAction={this.animateAction}
                    action={this.state.action}
                />
            </CardContainer >
        );
    }
}

export default Card;