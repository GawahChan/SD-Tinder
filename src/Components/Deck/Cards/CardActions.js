import React from 'react';

import {CardButtonContainer, CardLikedButton, CardDislikedButton, CardSuperlikedButton} from './style';

function CardActions({id, receiveAction}) {
    return (
        <CardButtonContainer>
            <CardDislikedButton onClick={() => receiveAction(id, 'disliked')}/>
            <CardSuperlikedButton onClick={() => receiveAction(id, 'superliked')} />
            <CardLikedButton onClick={() => receiveAction(id, 'liked')} />
        </CardButtonContainer>
    );
}

export default CardActions;