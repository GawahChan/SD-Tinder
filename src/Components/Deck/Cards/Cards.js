import React from "react";
import PropTypes from "prop-types";
import Card from "./Card";

import { CardsContainer } from "./style";

function Cards({ stuntDoubleList, handleAction }) {
  return (
    <CardsContainer>
      {stuntDoubleList.map(stuntDouble => {
        return (
          <Card
            key={stuntDouble.id}
            stuntDouble={stuntDouble}
            handleAction={handleAction}
          />
        );
      })}
    </CardsContainer>
  );
}

Cards.propTypes = {
  stuntDoubleList: PropTypes.array,
  handleAction: PropTypes.func
};

export default Cards;
