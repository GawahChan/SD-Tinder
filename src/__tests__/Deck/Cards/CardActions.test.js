import React from 'react';
import { shallow } from 'enzyme';
import CardActions from '../../../Components/Deck/Cards/CardActions';
import {CardLikedButton, CardDislikedButton, CardSuperlikedButton} from '../../../Components/Deck/Cards/style';


describe('CardActions component', () => {
    let mockReceiveAction = jest.fn()
    let props = { id: 0, receiveAction: mockReceiveAction }
    let cardActions = shallow(<CardActions {...props} />)


    it('renders correctly', () => {
        expect(cardActions).toMatchSnapshot();
    });

    it('renders three buttons, one action type each', () => {
        expect(cardActions.children()).toHaveLength(3);

        expect(cardActions.find(CardLikedButton)).toHaveLength(1);
        expect(cardActions.find(CardDislikedButton)).toHaveLength(1);
        expect(cardActions.find(CardSuperlikedButton)).toHaveLength(1);
    });

    describe('onTap buttons', () => {
        let id = 0;

        beforeEach(() => {
            mockReceiveAction.mockClear()
        });

        it('disliked button calls `receiveAction` prop', () => {
            let action = 'disliked'

            cardActions.find(CardDislikedButton).props().onTap();

            expect(mockReceiveAction).toHaveBeenCalledTimes(1);
            expect(mockReceiveAction).toHaveBeenCalledWith(id, action);
        });
        
        it('superliked button calls `receiveAction` prop', () => {
            let action = 'superliked'

            cardActions.find(CardSuperlikedButton).props().onTap();

            expect(mockReceiveAction).toHaveBeenCalledTimes(1);
            expect(mockReceiveAction).toHaveBeenCalledWith(id, action);
        });

        it('liked button calls `receiveAction` prop', () => {
            let action = 'liked'

            cardActions.find(CardLikedButton).props().onTap();

            expect(mockReceiveAction).toHaveBeenCalledTimes(1);
            expect(mockReceiveAction).toHaveBeenCalledWith(id, action);
        });
    })

});