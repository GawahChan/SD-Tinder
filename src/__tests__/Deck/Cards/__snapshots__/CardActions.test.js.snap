// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardActions component renders correctly 1`] = `
<styled.div>
  <Styled(Styled(Component))
    onClick={[Function]}
  />
  <Styled(Styled(Component))
    onClick={[Function]}
  />
  <Styled(Component)
    onClick={[Function]}
  />
</styled.div>
`;
